### ISSUE TITLE
exception 

### ISSUE THREAD

**Issue by mccraigmccraig** [clj-kondo/clj-kondo#1922]
**version**: v2022.12.10, installed via homebrew

**macro usage**: no

**platform**: macosx 12.5.1, emacs 29.0.50, flycheck, clj-kondo 2022.12.10 from homebrew

**editor**: emacs 29.0.50, flycheck-clj-kondo pinned at "d8a6ee9a16aa24b5be01f1edf9843d41bdc75555"

**problem**

```
qqu0upp:9:27:➜  promisespromises git:(nustreams) ✗ % clj-kondo --lint test/prpr/stream/cross_test.clj
Unexpected error. Please report an issue.
java.lang.ClassCastException: clojure.lang.PersistentArrayMap cannot be cast to clojure.lang.Named
	at clojure.core$namespace.invokeStatic(core.clj:1599)
	at clj_kondo.impl.types$nilable_QMARK_.invokeStatic(types.clj:68)
	at clj_kondo.impl.types$unnil.invokeStatic(types.clj:74)
	at clj_kondo.impl.types$match_QMARK_.invokeStatic(types.clj:120)
	at clj_kondo.impl.types$lint_map_types_BANG_.invokeStatic(types.clj:416)
	at clj_kondo.impl.types$lint_map_BANG_.invokeStatic(types.clj:428)
	at clj_kondo.impl.types$lint_arg_types.invokeStatic(types.clj:461)
	at clj_kondo.impl.linters$lint_arg_types_BANG_.invokeStatic(linters.clj:145)
	at clj_kondo.impl.linters$lint_var_usage.invokeStatic(linters.clj:458)
	at clj_kondo.core$run_BANG_.invokeStatic(core.clj:206)
	at clj_kondo.main$main.invokeStatic(main.clj:139)
	at clj_kondo.main$main.doInvoke(main.clj:135)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clj_kondo.main$_main$fn__13145.invoke(main.clj:186)
	at clj_kondo.main$_main.invokeStatic(main.clj:186)
	at clj_kondo.main$_main.doInvoke(main.clj:184)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clj_kondo.main.main(Unknown Source)
```

**repro**

```
git clone https://github.com/yapsterapp/promisespromises.git ; cd promisespromises ; git checkout 5956eb7300fbf5622a6f57d2e69ec7dab5be98a9 

clj-kondo --lint test/prpr/stream/cross_test.clj
```

**config**

the bug is related to the config.edn generated by malli.dev - `rm .clj-kondo/metosin/malli-types/config.edn` before linting makes the issue go away. the full .clj-kondo config is included in the repo above

**expected behavior**

- not throw an exception 😀
- i'm guessing this could also be a problem with the `config.edn` that `malli.dev` is generating - but in that case clj-kondo should at least give some indication of what's wrong with the `config.edn`

i'm happy to help with a PR or further investigation if that's useful - drop me a comment

**Comment by borkdude** [comment]
Thanks!

**Comment by borkdude** [comment]
Yes, narrowing down the config would help and then coming up with a minimal piece of code which triggers it.

**Comment by mccraigmccraig** [comment]
ok, i'll do some binary-chopping on that repo and see where i get to...

**Comment by mccraigmccraig** [comment]
ok, chopping done - here's a minimal repro:

```
git clone https://github.com/mccraigmccraig/clj-kondo-malli-error-2022-12-20.git
cd clj-kondo-malli-error-2022-12-20
clj-kondo --lint src/bar.cljc
```

and to show that it's the malli-generated `config.edn` that's at fault:

```
rm .clj-kondo/metosin/malli-types/config.edn
clj-kondo --lint src/bar.cljc
```

**Comment by borkdude** [comment]
The error in the outputted spec is here:

```
[{:op :keys,
  :req {:keys {:op :rest, :spec :keyword}}}
 :any]
```

I don't think `:req` is correct, but first let me try to understand the malli spec here:

```
[:map [:keys [:+ :keyword]]]
```

What is this supposed to mean?

**Comment by mccraigmccraig** [comment]
it's for a map with a `:keys` entry with a vector of one or more keywords

```
(malli.core/validate [:map [:keys [:+ :keyword]]] {:keys []}) => false
(malli.core/validate [:map [:keys [:+ :keyword]]] {:keys [:foo]}) => true
(malli.core/validate [:map [:keys [:+ :keyword]]] {:keys [:foo :bar]}) => true
```

**Comment by borkdude** [comment]
aha! yeah, this makes sense, I'll take a look

**Comment by borkdude** [comment]
I dug into this a little deeper and my conclusion is that malli uses `{:op :rest}` the wrong way. `{:op :rest}` should only be used to describe the types of varargs arguments, not as a way to describe the contents of a map key (e.g. a vector of keywords). So I think it would be best if malli didn't generate this. I'll fix the type checking to never throw though, since that's not desirable.

cc @ikitommi
